plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -996,6 +996,12 @@ public Answer backupSnapshot(final CopyCommand cmd) {
                             primaryStore.getUuid());
                     if (state == DomainInfo.DomainState.VIR_DOMAIN_RUNNING && !primaryStorage.isExternalSnapshot()) {
                         final DomainSnapshot snap = vm.snapshotLookupByName(snapshotName);
+                        try {
+                            vm.suspend();
```

Comment:
When we need to delete a snapshot, is it a KVM/libvirt requirement to suspend the domain? Could it be something to do with old/archaic libvirt/qemu-img version?

Reply:
Hello,
It's not required by the documentation, but many users encountered image corruption, so it seems to be a good idea to suspend and resume the VM while deleting a snapshot (Kevin Wolf from RedHat mentioned it too: https://bugzilla.redhat.com/show_bug.cgi?id=920020#c5).
Thanks.

Reply:
Thanks @melnik13 I'll review/test this.